--- gstreamer-0.10.35/gst/gstinfo.c.orig	2011-06-14 07:36:48.000000000 -0600
+++ gstreamer-0.10.35/gst/gstinfo.c	2011-09-16 13:54:20.000000000 -0600
@@ -334,6 +334,7 @@
   /* get time we started for debugging messages */
   _priv_gst_info_start_time = gst_util_get_timestamp ();
 
+#ifdef GST_USING_PRINTF_EXTENSION
 #ifdef HAVE_PRINTF_EXTENSION
 #ifdef HAVE_REGISTER_PRINTF_SPECIFIER
   register_printf_specifier (GST_PTR_FORMAT[0], _gst_info_printf_extension_ptr,
@@ -347,6 +348,7 @@
       _gst_info_printf_extension_segment, _gst_info_printf_extension_arginfo);
 #endif
 #endif
+#endif
 
   /* do NOT use a single debug function before this line has been run */
   GST_CAT_DEFAULT = _gst_debug_category_new ("default",
@@ -354,7 +356,7 @@
   _GST_CAT_DEBUG = _gst_debug_category_new ("GST_DEBUG",
       GST_DEBUG_BOLD | GST_DEBUG_FG_YELLOW, "debugging subsystem");
 
-  gst_debug_add_log_function (gst_debug_log_default, NULL);
+//  gst_debug_add_log_function (gst_debug_log_default, NULL);
 
   /* FIXME: add descriptions here */
   GST_CAT_GST_INIT = _gst_debug_category_new ("GST_INIT",
diff -Naur gstreamer-0.10.35-orig/gst/gstplugin.c gstreamer-0.10.35/gst/gstplugin.c
--- gstreamer-0.10.35-orig/gst/gstplugin.c	2009-01-06 13:59:21.000000000 -0700
+++ gstreamer-0.10.35/gst/gstplugin.c	2011-06-23 11:37:50.760237053 -0600
@@ -744,14 +744,14 @@
     goto return_error;
   }
 
-  if (g_stat (filename, &file_status)) {
-    GST_CAT_DEBUG (GST_CAT_PLUGIN_LOADING, "problem accessing file");
-    g_set_error (error,
-        GST_PLUGIN_ERROR,
-        GST_PLUGIN_ERROR_MODULE, "Problem accessing file %s: %s", filename,
-        g_strerror (errno));
-    goto return_error;
-  }
+//  if (g_stat (filename, &file_status)) {
+//    GST_CAT_DEBUG (GST_CAT_PLUGIN_LOADING, "problem accessing file");
+//    g_set_error (error,
+//        GST_PLUGIN_ERROR,
+//        GST_PLUGIN_ERROR_MODULE, "Problem accessing file %s: %s", filename,
+//        g_strerror (errno));
+//    goto return_error;
+//  }
 
   flags = G_MODULE_BIND_LOCAL;
   /* libgstpython.so is the gst-python plugin loader. It needs to be loaded with
--- gstreamer-0.10.35/plugins/elements/gstqueue.c.orig	2011-06-13 16:35:33.000000000 -0600
+++ gstreamer-0.10.35/plugins/elements/gstqueue.c	2011-11-22 08:35:53.392625000 -0700
@@ -1138,7 +1138,7 @@
       queue->head_needs_discont = FALSE;
     }
 
-    caps = GST_BUFFER_CAPS (buffer);
+    caps = gst_buffer_get_caps(buffer);
 
     GST_QUEUE_MUTEX_UNLOCK (queue);
     /* set the right caps on the pad now. We do this before pushing the buffer
@@ -1146,8 +1146,13 @@
      * be set on the pad, which might fail because this will be propagated
      * upstream. Also note that if the buffer has NULL caps, it means that the
      * caps did not change, so we don't have to change caps on the pad. */
-    if (caps && caps != GST_PAD_CAPS (queue->srcpad))
-      gst_pad_set_caps (queue->srcpad, caps);
+    if (NULL != caps)
+    {
+        if (caps != GST_PAD_CAPS (queue->srcpad))
+          gst_pad_set_caps (queue->srcpad, caps);
+
+        gst_caps_unref (caps);
+    }
 
     if (queue->push_newsegment) {
       gst_queue_push_newsegment (queue);
--- ./gstreamer-0.10.35/libs/gst/base/gstbasesink.c.orig	2012-04-10 07:41:52.000000000 -0600
+++ ./gstreamer-0.10.35/libs/gst/base/gstbasesink.c	2012-04-10 07:42:58.000000000 -0600
@@ -2861,7 +2861,7 @@
     if (GST_CLOCK_TIME_IS_VALID (priv->last_render_time) &&
         rstart - priv->last_render_time > GST_SECOND) {
       late = FALSE;
-      GST_ELEMENT_WARNING (basesink, CORE, CLOCK,
+      GST_ELEMENT_INFO (basesink, CORE, CLOCK,
           (_("A lot of buffers are being dropped.")),
           ("There may be a timestamping problem, or this computer is too slow."));
       GST_CAT_DEBUG_OBJECT (GST_CAT_PERFORMANCE, basesink,
